Specifications/monitoring.md

```diff
@@ -0,0 +1,308 @@
+| `player_timestamp` | The current player timestamp, as retrieved from the playlist. Omitted if not available | [Unix timestamp](https://unixtime.org) in milliseconds | `1717665997932` |
+| `duration` | The content duration, as retrieved from the playlist | Offset from the beginning of the content, in milliseconds | `16548` |
+| `severity` | The error severity | `WARNING`, `FATAL` | `WARNING` |
+| `url` | The URL that was affected by the error | String | `https://rts1-lsvs.akamaized.net/out/v1/62441d2399f14dce9e558b5503edba11/index_1_948290.ts` |
+
+> [!IMPORTANT]
+> Requirements for each key are not provided explicitly but implementations should fill as much information as possible. If some information cannot be reliably determined it should be omitted.
+
+Some remarks:
+
+- If the error occurs before playback has started (e.g. during metadata retrieval) then `player_position`, `player_timestamp` and `duration` must be omitted.
+- The `url` should describe the content that was affected as closely as possible, down to media playlists or segment URLs, provided this information is available.
+- The `log` is informally defined so that any useful information can be added for investigation purposes.
+
+### Example
+
+```json
+{
+    "event_name": "ERROR",
+    "session_id": "0aef9cb4-5b0f-47eb-9926-a799abb9332f",
+    "timestamp": 1723201568953,
+    "version": 1,
+    "data": {
+        "message": "Segment exceeds specified bandwidth for variant",
+        "name": "CoreMediaErrorDomain(-12318)",
+        "player_position": 724,
+        "severity": "WARNING",
+        "url": "https:\/\/rts-vod-amd.akamaized.net\/ww\/14773895\/5dc2e58b-6c38-385a-a674-259827b521e8\/index-f3-v1.m3u8"
+    }
+}
+```
+
+## Status Event `data`
+
+Other events must be sent during the playback session:
+
+| Event name | Description | Time at which the event is sent |
+| - | - | - |
+| `STOP` | Playback ended | When playback ends normally or is interrupted |
+| `HEARTBEAT` | Informs that the session is still alive | Every 30 seconds (also when paused). First heartbeat sent right after `START` |
+
+The associated event data dictionary supports the following keys:
+
+| Key | Description | Format | Examples |
+| - | - | - | - |
+| `airplay` | A value indicating whether AirPlay is currently active | Boolean | `true` |
+| `bandwidth` | Bandwidth | Number in bits per second | `4000000` |
+| `bitrate` | Bitrate of the content being played | Number in bits per second | `1000000` |
+| `buffered_duration` | Duration of the content currently available in buffer | Duration in milliseconds | `12000` |
+| `playback_duration` | The duration of the playback session | Duration in milliseconds | `40000` |
+| `player_position` | The current player position | Duration in milliseconds | `16548` |
+| `stall` | Stall information | JSON dictionary | `{ ... }` |
+| `stream_type` | Stream type | `on-demand`, `live` | `on-demand` |
```

Comment:
@defagos do you think it's worth mentioning why the `stream_type` property has been placed at this level rather than in the `start` event? I'm pointing this out because it may seem counter-intuitive, but there's a real reason for it. In case anyone outside the team ever wonders.

Reply:
Sure, will do, thanks.

Reply:
Done. I am closing this conversation, feel free to open a new one if you think the current explanation can be improved.